Use C4 diagrams for documentation

hannes-mk opened the pull request:
Document the existing state with C4 diagrams, cf. https://c4model.com/
Support of C4 in mermaid is marked "experimental", but seems to work so far, cf.  http://mermaid.js.org/syntax/c4.html

hannes-mk committed:
Add initial diagram

- BmJ with two offices and three documentalists using their web browser
- Web app with frontend, backend and relational database

RISDEV-0000

hannes-mk committed:
Re-order to three columns

RISDEV-0000

hannes-mk reviewed (commented):
The current version is not the context diagram, but too detailed.

There should be the BmJ with the Documentalists and DS with the Web App. Nothing more.

hannes-mk committed:
Split into system context and container diagram

RISDEV-0000

hannes-mk committed:
Move diagrams to new "architecture" folder

RISDEV-0000

VictorDelCampo commented on doc/architecture/c4-system-context.md:
```diff
@@ -0,0 +1,35 @@
+    System("portal", "Portal", "Gives the public access to norms")
+    Rel("ris-norms", "portal", "Makes new norms accessible to the portal")
+
+    Person("public", "General public, lawyer, law maker,<br> public servant, private enterprise...", "Searches, reads or processes norms for<br> private or professional reasons")
```
Why this "Person"? it is not relevant for the architecture of norms. As soon as we send data (LDML.de) to portal our system context is done. What users/stakeholders are interacting with the portal is not relevant for the norms architecture

hannes-mk replied:
I agree. That's the scope of the portal team and we should not make assumptions about it in our docs.
Will update.

hannes-mk committed:
Rename folder to "architecture-diagrams"

- Use Person_Ext and System_Ext
- Remove user of the portal as that's out of our scope

RISDEV-0000

hannes-mk committed:
Use Person_Ext, add portal as Container_Ext

RISDEV-0000

hannes-mk committed:
Reference architecture diagrams in README

RISDEV-0000
